### DISCUSSION TITLE
How should we generate pdfs?

### DISCUSSION THREAD

**Discussion by aayushchugh** [keizerworks/invoicen#14] [General]
Need some optimised solutions for generating PDFs in react/nextjs

**Comment by Sudarsh1010** [comment]
I understood the reason behind generating pdf in client side. but In future, If gonna implement logging, history and all then pdf generation logic should move to server-side? so public user will get pdf in response and member's pdf will be stored in some sort of bucket, URL for the file will be send in response? I guess we can discuss more on this topic to get some clear view on this topic.

**Reply by aayushchugh** [reply]
Yup that's what we'll do. we'll save pdfs of logged in users in buckets

**Comment by aayushchugh** [comment]
I am using react-pdf for this
